### DISCUSSION TITLE
Difficulties following along on tutorial

### DISCUSSION THREAD

**Discussion by WillAdams** [dune3d/dune3d#118] [General]
I am trying to follow along at:

https://docs.dune3d.org/en/latest/simple-part.html

(for convenience I have that open in the web browser on my Kindle Scribe, so seeing it in black and white)

I launch the app and get:

![image](https://github.com/dune3d/dune3d/assets/3543154/2644979d-25c8-4647-a10c-78251fd94afe)

Should the tutorial mention the need to click "Create a New document" and maximize or adjust window size?

I believe I was able to turn off the other workplanes as suggested and then get back to the XY plane for Sketching:

![image](https://github.com/dune3d/dune3d/assets/3543154/afe42d74-4772-4ee3-9ba9-b08a336537ac)

and drawing the outline seemed to work as expected:

![image](https://github.com/dune3d/dune3d/assets/3543154/82af996e-f903-4467-a249-31f10b59a8b7)

The problems began when I selected the nodes to constrain things:

![image](https://github.com/dune3d/dune3d/assets/3543154/06a65f2b-ef12-4555-a321-4fae2ce62e0f)

![image](https://github.com/dune3d/dune3d/assets/3543154/7f91fc2f-8bc6-48ec-be99-eeb1152b0413)

The first few times I tried that I was unable to do it successfully --- once I got a circle --- not sure what the error I was making was, but perhaps possible difficulties could be mentioned?

The comment is made:

>This should have removed two degrees of freedom. 

Where are these? How are they identified? How does one count them up and keep track?

>In the workspace browser on the left side of the window, we should now see that our sketch has 4 degrees of freedom. 

Which 4 things indicate degrees of freedom?

![image](https://github.com/dune3d/dune3d/assets/3543154/e5fccd02-c05d-421b-bf37-a99a401e9afe)

I believe I managed to get the vertical point placed:

![image](https://github.com/dune3d/dune3d/assets/3543154/bb9ecc61-ebea-41ab-aae2-e8d065c670f7)

but when I tried to select it and one of the lower ones as directed I had trouble because I shift-clicked out of habit because that is how multiple selections are made in most graphical tools --- hitting esc and just clicking of course worked.

>With the 2D sketch being fully constrained, we can move on to make it 3D.

Is the sketch fully constrained?

![image](https://github.com/dune3d/dune3d/assets/3543154/ca3892da-8a93-4567-abc5-61ec79a7e544)

If it is, I can't tell....

>You should now have a workplane that looks like this. Make it this sketch’s active workplane by selecting “Set workplane” from its context menu.

The context menu I'm getting:

![image](https://github.com/dune3d/dune3d/assets/3543154/d962e1b1-0838-4e6c-a7b9-c42ca356d581)

has "Set active workplane" rather than "Set workplane"

and when that was done, the rotation was off:

![image](https://github.com/dune3d/dune3d/assets/3543154/0b578ac1-a9ce-4fff-9e69-ca8e8207c26b)

Instructions on how to adjust that? Or modify the instructions to ensure it doesn't happen?

Tried using the Rotate command to get the same appearance:

![image](https://github.com/dune3d/dune3d/assets/3543154/b6b86ce8-39b9-44ea-b429-caaba6cdf483)

but couldn't make a change stick.

>To remove all degrees of freedom:
> - Constrain the construction circle’s diameter
> - Constrain the bottom line of the hexagon to be horizontal
> - Constrain the horizontal and vertical distance from the top-left point of the face

It took me a couple of tries to figure this out --- perhaps add parenthetical hints? (select circle and right-click and choose the diameter constraint), (hit esc and then select the two points and apply the constraint).

When I tried to apply the extrusion I got:

![image](https://github.com/dune3d/dune3d/assets/3543154/ceca80ee-24ae-4492-b8f1-9bd900aef3d5)

(after a bit of dragging/rotating)

which is obviously wrong and even indicates the error I made --- how to recover? 

I will sleep on all this and give it another go from the beginning in the morning, but I hope that the above helps to show what a naïve user will experience when first trying things out --- should there be an atlas/overview/interface guide which shows all the screens and their constituent parts and explains the meaning and significance of each?

**Comment by WillAdams** [comment]
Managed to replicate the initial "Constrain Coincident" creating a circle:

![image](https://github.com/dune3d/dune3d/assets/3543154/1401d93a-865e-4adc-85bf-80d81782632f)

Previous state (after undoing) was:

![image](https://github.com/dune3d/dune3d/assets/3543154/3f273e55-861a-493f-a9bd-ae616d33df47)

and no matter what I do (hitting esc, changing selection order, &c.) when I choose:

![image](https://github.com/dune3d/dune3d/assets/3543154/4e927292-de2c-4452-85cf-3eda513372d6)

I get a circle.

Attached here.

[tut_1_shape_becomes_circle..d3ddoc.zip](https://github.com/user-attachments/files/15752706/tut_1_shape_becomes_circle.d3ddoc.zip)

**Reply by WillAdams** [reply]
and now I'm getting a semi-circle/chord:

![image](https://github.com/dune3d/dune3d/assets/3543154/6848ceee-37d7-4932-845b-c0e3da45db0a)

[tut_1_shape_becomes_semicircle.zip](https://github.com/user-attachments/files/15752730/tut_1_shape_becomes_semicircle.zip)

**Comment by holgerlembke** [comment]
Might it be the same thing I encountered?

https://github.com/dune3d/dune3d/discussions/91

**Reply by WillAdams** [reply]
Good question --- maybe? I'm using the current 1.1 release on Windows (which I guess I should have mentioned).

I'll continue experimenting until I'm successful (or bored, or frustrated).

I will note that I'm getting much further with Dune 3D, and it's feeling far more comfortable than any other 3D CAD tool I've tried --- heretofore I've only been successful with OpenSCAD (and 2.5/3D CAM tools such as MeshCAM and Carbide Create).

**Reply by holgerlembke** [reply]
I think we need to wait for some more releases... As far as I see, it is in a real good state after only two releases. Now it needs some time (and developers) to root out the errors.

I'm with you with OpenSCAD... I clearly love it (https://github.com/holgerlembke/OpenSCAD-Modelle) but I miss a lot of advanced features like roundings and chamfers "the easy way".

**Reply by WillAdams** [reply]
Looking forward to a new release --- but until then, I hope that I am able to assist with documentation and UI clarity/consistency --- I'm making progress, and I would hope that making the tutorials clearer will help to attract new users.

**Comment by WillAdams** [comment]
Another thing to mention in the tutorial when using the small parameter windows:

![image](https://github.com/dune3d/dune3d/assets/3543154/9121c664-0718-4930-b768-f521d27d06cc)

is that pressing enter twice will 1 apply, 2 close the window (which is _very_ nice UI)

**Reply by WillAdams** [reply]
Got stuck on applying the constraint to the center arc point:

![image](https://github.com/dune3d/dune3d/assets/3543154/339bbfef-4c68-4322-9e85-09728d71f26c)

Not sure how to proceed/fix --- here's the file:

[tut_1_shape_arc_center_will_not_constrain.zip](https://github.com/user-attachments/files/15752785/tut_1_shape_arc_center_will_not_constrain.zip)

Will start over on the tutorial and hopefully manage to (again) draw something which will let me proceed).

**Reply by WillAdams** [reply]
At:

>Use the “Constrain vertical” tool to constrain it to be directly above the arc’s center.

I'm stuck for the third or fourth time --- there doesn't seem to be a "Constrain vertical" tool:

![image](https://github.com/dune3d/dune3d/assets/3543154/8b7d2464-1600-415d-8d95-3122f76f2b2d)

and right-clicking still doesn't yield a menu which indicates this is constrained.

Here's this state in case it's meaningfully different:

[tut_1_shape_arc_center_will_not_constrain_notool.zip](https://github.com/user-attachments/files/15752928/tut_1_shape_arc_center_will_not_constrain_notool.zip)

**Comment by WillAdams** [comment]
Kept going and was able to assign the vertical distance constraint and so forth and arrived at:

>You should now have a workplane that looks like this. Make it this sketch’s active workplane by selecting “Set workplane” from its context menu.

![image](https://github.com/dune3d/dune3d/assets/3543154/c2dd0a5f-7ecf-4390-aeed-1ddb2ed232ba)

Should that be "Set active workplane"? 

EDIT: and I see that this was pointed out above when I previously arrived at this point --- my apologies for the redundancy.

**Comment by WillAdams** [comment]
I had problems getting the chamfer to show up --- had to toggle "show body" on/off, but, success!

![image](https://github.com/dune3d/dune3d/assets/3543154/9ddf9847-b60c-4ab9-8e41-49542a2c43f1)

**Comment by WillAdams** [comment]
Trying again now that 1.4 has been released --- still running on Windows 11 on a Samsung Galaxy Book 3 Pro 360.

Had difficulty setting the vertical constraint again because it wasn't clear to me that two points needed to be selected at the same time.

Setting the vertical distance constraint had a markedly different appearance:

<img width="2880" height="1728" alt="image" src="https://github.com/user-attachments/assets/56c7fce9-65be-4a56-bde8-2cc3dc041c5c" />

Is that as expected in 1.4?

Still mystified as to how one counts that there are zero degrees of freedom...

Also, Extrude didn't work for me:

<img width="2880" height="1728" alt="image" src="https://github.com/user-attachments/assets/bbffb08e-edc6-4c02-aca6-be88242009ad" />

Is there information which could be added to the tutorial which would make this clearer? Concepts which should be explained rather than assumed? Troubleshooting steps which might be added?

**Comment by carrotIndustries** [comment]
> Setting the vertical distance constraint had a markedly different appearance:

looks you picked "distance" instead of "vertical distance"

> Also, Extrude didn't work for me

Well, extrude only works on closed paths and yours obviously isn't.

**Reply by WillAdams** [reply]
Tried to make some minor corrections in the docs, but obviously didn't get it right at first --- I think I puzzled it out, hope it helps!

**Comment by WillAdams** [comment]
Thanks! Are these things which the tutorial should be cautioned about?

In particular, how does one determine if a path is closed or open? If one ends up with an open path is there a way to fix this?

**Comment by awe-s0me** [comment]
Hi, I am trying to learn Dune3D and I also have had troubles with a point not knowing if it sits on the top of the arc. But I've figured it out.
1. you need to have the latest version of Dune3D
2. In the Height paragraph the tutorial is missing that you need to select A) the reference point on the arc and B) the Arc in workplane (center) to align. 
<img width="310" height="428" alt="obrazek" src="https://github.com/user-attachments/assets/ed4b3a79-a730-434a-9274-f8d16cfed540" />

3. then Right click on one of these points and select Constrain vertical (auto)
<img width="311" height="391" alt="obrazek" src="https://github.com/user-attachments/assets/5884a6ed-b636-4728-9a55-8ff76ad4109d" />

4. you know then that it is aligned when there is small violet line (in the orange circle on the picture below)
<img width="272" height="400" alt="obrazek" src="https://github.com/user-attachments/assets/d1e6a366-bba4-464e-95a9-8397f34b117d" />

Hope it helps.

**Reply by WillAdams** [reply]
Thanks!

I'll try to find the time to work through this again and see if I can manage to write up something which makes sense to me and passes muster for a pull/push request (I find Git's terminology confusing).

**Comment by bilal333kcm-arch** [comment]
Hi everyone,

I’m currently working on a project where I need to design and analyze an aircraft fuselage section using ANSYS, and I’d like to integrate Python (PyAnsys) into the workflow.

I’m still a beginner with PyAnsys, so I’m looking for guidance on how to get started properly — especially:

Setting up geometry and meshing with scripts
Running structural analysis using Python
Best practices or tutorials for aerospace applications

If anyone has experience with this or can share resources, example scripts, or advice, I’d really appreciate it.

Thanks in advance! 🙏
<img width="960" height="423" alt="fuselage" src="https://github.com/user-attachments/assets/61e2eeb8-169d-45be-839d-b659c3104a7d" />
